x-pack/plugins/elastic_assistant/server/routes/user_conversations/update_route.ts

```diff
@@ -54,14 +54,14 @@ export const updateConversationRoute = (router: ElasticAssistantPluginRouter) =>
             });
           }
 
-          const dataClient = await ctx.elasticAssistant.getAIAssistantConversationsDataClient();
           const authenticatedUser = ctx.elasticAssistant.getCurrentUser();
           if (authenticatedUser == null) {
             return assistantResponse.error({
               body: `Authenticated user not found`,
               statusCode: 401,
             });
           }
+          const dataClient = await ctx.elasticAssistant.getAIAssistantConversationsDataClient();
```

Comment:
Just an FYI, but I recently added a new `performChecks()` helper for doing the authenticatedUser, license and capability checks in effort to clean up these route definitions. It's available here: https://github.com/elastic/kibana/blob/f8bce6c9d116341ba1f7d339ee69320ea7b2ae9c/x-pack/plugins/elastic_assistant/server/routes/helpers.ts#L627-L639